src/overlays/actors/ovl_En_Rail_Skb/z_en_rail_skb.c

```diff
@@ -101,122 +158,1003 @@ static DamageTable D_80B7346C = {
+    }
+}
+
+void func_80B71488(EnRailSkb* this) {
+    this->unk_402 |= 0x40;
+    this->actor.flags &= ~1;
+    if (this->unk_2E0 != 0) {
+        this->unk_2E4 = this->unk_2E0 - 1;
+    } else {
+        this->unk_2E4 = this->unk_2E8 - 1;
+    }
+    this->actionFunc = func_80B714D8;
+}
+
+void func_80B714D8(EnRailSkb* this, GlobalContext* globalCtx) {
+    if (this->actor.draw != NULL) {
+        this->actor.draw = NULL;
+    }
+
+    if (this->unk_2E0 == this->unk_2E4) {
+        func_80B7151C(this);
+    }
+}
+
+void func_80B7151C(EnRailSkb* this) {
+    this->actor.shape.yOffset = -5000.0f;
+    this->actor.colChkInfo.health = 2;
+    this->unk_402 = 0;
+    this->actor.flags |= 1;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALKID_APPEAR);
+    this->actor.draw = EnRailSkb_Draw;
+    this->actor.shape.shadowAlpha = 0;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    func_800BDC5C(&this->skelAnime, sAnimations, 1);
+    this->actionFunc = func_80B715AC;
+}
+
+void func_80B715AC(EnRailSkb* this, GlobalContext* globalCtx) {
+    if (Math_SmoothStepToF(&this->actor.shape.yOffset, 0.0f, 0.5f, 500.0f, 10.0f) == 0.0f) {
+        func_80B70FA0(this);
+    } else {
+        func_80B72430(this, globalCtx, 1);
+    }
+
+    if (this->actor.shape.shadowAlpha != 255) {
+        if (this->actor.shape.shadowAlpha >= 235) {
+            this->actor.shape.shadowAlpha = 255;
+        } else {
+            this->actor.shape.shadowAlpha += 20;
+        }
+    }
+}
+
+void func_80B71650(EnRailSkb* this) {
+    this->unk_3FE = 0;
+    if (this->actionFunc != func_80B70FF8) {
+        func_800BDC5C(&this->skelAnime, sAnimations, 0);
+    }
+    this->actionFunc = func_80B716A8;
+}
+
+void func_80B716A8(EnRailSkb* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->unk_3FE = 1;
+        func_80B71D8C(this, globalCtx, func_80B723F8);
+        if (!func_80B7285C(this)) {
+            func_801518B0(globalCtx, 5100, &this->actor);
+            this->unk_400 = 5100;
+            func_80B72830(this, 1);
+        } else {
+            func_801518B0(globalCtx, 5109, &this->actor);
+            this->unk_400 = 5109;
+        }
+        func_800BDC5C(&this->skelAnime, sAnimations, 12);
+        func_80B717C8(this);
+    } else if ((this->actor.xzDistToPlayer < 100.0f) && !(this->collider.base.acFlags & AC_HIT)) {
+        func_800B8614(&this->actor, globalCtx, 100.0f);
+    }
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 1, 0x71C, 0xB6);
+}
+
+void func_80B717C8(EnRailSkb* this) {
+    this->unk_3F2 = 0;
+    this->actionFunc = func_80B717E0;
+}
+
+void func_80B717E0(EnRailSkb* this, GlobalContext* globalCtx) {
+    this->unk_3FA = 0;
+
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+            break;
+
+        case 3:
+            if ((globalCtx->gameplayFrames & 1) != 0) {
```

Comment:
Would this match with a `% 2`
```suggestion
            if ((globalCtx->gameplayFrames % 2) != 0) {
```